app/code/Magento/Review/Observer/CatalogBlockProductListCollectionInitializeObserver.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Review\Observer;
+
+use Magento\Framework\Event\ObserverInterface;
+
+class CatalogBlockProductListCollectionInitializeObserver implements ObserverInterface
+{
+    /**
+     * Store manager
+     *
+     * @var \Magento\Store\Model\StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @param \Magento\Store\Model\StoreManagerInterface $storeManager
+     */
+    public function __construct(
+        \Magento\Store\Model\StoreManagerInterface $storeManager
+    ) {
+        $this->storeManager = $storeManager;
+    }
+
+    /**
+     * Join review summary to collection
+     *
+     * @param \Magento\Framework\Event\Observer $observer
+     * @return $this
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function execute(\Magento\Framework\Event\Observer $observer)
+    {
+        $productCollection = $observer->getEvent()->getCollection();
+        if ($productCollection instanceof \Magento\Catalog\Model\ResourceModel\Product\Collection && !$productCollection->isLoaded()) {
+            $reviewEntitySubSelect = $productCollection->getConnection()->select();
+            $reviewEntitySubSelect
+                ->from(
+                    ['review_entity' => $productCollection->getTable('review_entity')],
+                    ['entity_id']
+                )->where(
+                    'entity_code = ?',
+                    \Magento\Review\Model\Rating::ENTITY_PRODUCT_CODE
+                );
+            $currentStore = $this->storeManager->getStore()->getId();
+            $joinCond = new \Zend_Db_Expr("e.entity_id = review_summary.entity_pk_value AND review_summary.store_id = {$currentStore} AND review_summary.entity_type = ({$reviewEntitySubSelect})");
+            $productCollection->getSelect()
+                ->joinLeft(
```

Comment:
Such code must be placed in some model, not in observer.

Also, I believe some adjustments should be made to `appendSummary` (for example, deprecation).

Reply:
@orlangur Code moved to model